Updating contributing readme file with additional instructions closes #476 (#477)
bob-crutchley/courseware: CONTRIBUTING.md

All courseware on this project is in small and digestable modules.
These modules are then organised into topics.
Moving forward, we will be introducing courses and pathways into the project.
- **Topics**
    - This is a subject, which could be a programming language or tool. For example, Java, Jenkins, Kubernetes and Google Cloud Platform would all be considered subjects.
- **Modules**
    - Modules are the topics broken down into much smaller, digestible content and tasks.
- **Courses**
    - A course is a carefully ordered set of modules, from any topics, that are relevant for the course outcome.
- **Pathways**
    - Pathways are courses that have been put in order.
## Naming Conventions

> Files, folders and packages should be in lowercase with exceptions to the `README.md` file.

Reviewer: ```suggestion
Files, folders and packages should be in lowercase, with the exception of the `README.md` file.
```

Quizzes can be made by using hidden drop downs for the answers.

## Review Process

> Modules will go through 2 sets of reviewing before being released into the *master* branch.

Reviewer: ```suggestion
Modules will go through 2 sets of reviewing before being released into the **master** branch.
```

> The first review is for technical and implementation purposes, therefore they would be reviewed ideally by someone with some knowledge in the area:

Reviewer: ```suggestion
The first review is for technical and implementation purposes, so this would ideally be reviewed by someone with some knowledge in the area:
```

- Are the tasks clear and solvable? (attempt the task)
- Are the statements and explained concepts correct throughout the module?
- Does it serve its purpose well, as a piece of material for learning?

> - Does it make sense grammatically, are there any spelling errors etc.

Reviewer: ```suggestion
- Does it make sense grammatically - are there any spelling errors, etc.?
```

> - Is the formatting correct

Reviewer: ```suggestion
- Check that the formatting is correct:
```

>     - Are the required headers there and used correctly

Reviewer: ```suggestion
    - Are the required headers there and used correctly?
```

>     - Are each sentence on a separate line

Reviewer: ```suggestion
    - Is each sentence on a separate line?
```

> - Is the module not too big or expands beyond what should be covered

Reviewer: ```suggestion
- Is the module too large or expanding beyond what should be covered?
```

> - Is the *pull request* without *merge conflicts*

Reviewer: ```suggestion
- Is the *pull request* without *merge conflicts*?
```

> - Is the *pull request* labeled with the correct **release label**

Reviewer: ```suggestion
- Is the *pull request* labeled with the correct **release label**?
```

> The second set of reviewing is more for grammatical purposes and general look of the material:

Reviewer: ```suggestion
The second set of reviewing is more for grammatical purposes and the general look of the material:
```

> - Any spelling mistakes.

Reviewer: Is this is for this one?
I'd be looing at formatting too , like headers not looking weird, bold and italic being used properly, table formatting, image placement, etc?

Author: I'll add those to it to make it more clear
